tests/modules/apigee_mock_client/src/MockHandlerStack.php

```diff
@@ -0,0 +1,164 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_mock_client;
+
+use Apigee\Edge\Exception\ClientErrorException;
+use Drupal\Core\Queue\QueueFactory;
+use GuzzleHttp\Handler\MockHandler;
+use GuzzleHttp\Psr7\Response;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Symfony\Component\Yaml\Yaml;
+
+class MockHandlerStack extends MockHandler {
+
+  /**
+   * Responses that have been loaded from the response file.
+   *
+   * @var array
+   */
+  protected $responses;
+
+  /**
+   * The twig environment for getting response data.
+   *
+   * @var \Twig_Environment
+   */
+  protected $twig;
+
+  /**
+   * @var \Drupal\Core\Queue\QueueInterface
+   */
+  protected $database_queue;
+
+  /**
+   * @param \Twig_Environment $twig
+   * @param \Drupal\Core\Queue\QueueFactory $queue_factory
+   */
+  public function __construct(\Twig_Environment $twig, QueueFactory $queue_factory) {
+    $this->twig = $twig;
+    $this->database_queue = $queue_factory->get('apigee_edge_mock_responses', TRUE);
+
+    parent::__construct();
+  }
+
+  /**
+   * Queue a response that is in the catalog. Dynamic values can be passed and
+   * will be replaced in the response. For example `['foo' => [':bar' => 'baz']]`
+   * will load the catalog entry named `foo `and replace `:bar` with "baz" in the
+   * body text.
+   *
+   * @param string|array $response_ids
+   *   The response id to queue in one of the following formats:
+   *     - `'foo'`
+   *     - `['foo', 'bar']` // Queue multiple responses.
+   *     - `['foo' => [':bar' => 'baz']]` // w/ replacements for response body.
+   *
+   * @return $this
+   *
+   * @throws \Twig_Error_Loader
+   * @throws \Twig_Error_Runtime
+   * @throws \Twig_Error_Syntax
+   */
+  public function queueFromResponseFile($response_ids) {
+    $org_name = \Drupal::service('apigee_edge.sdk_connector')->getOrganization();
+
+    if (empty($this->responses)) {
+      // Get the module path for this module.
+      $module_path = \Drupal::moduleHandler()->getModule('apigee_mock_client')->getPath();
+      $this->responses = Yaml::parseFile($module_path . '/response_catalog.yml')['responses'];
+    }
+    // Loop through responses and add each one.
+    foreach ((array) $response_ids as $index => $item) {
+      // The catalog id should either be the item itself or the keys if an associative array has been passed.
+      $id = !is_array($item) ? $item : $index;
+      // Body text can have elements replaced in it for certain values. See: http://php.net/strtr
```

Comment:
Is this an out of date comment saying `See: http://php.net/strtr`?

Reply:
Resolved in commit f134dd7.